lib/new-admin/graphql/modules/userManagement.js

```diff
@@ -0,0 +1,228 @@
+    return user
+  }).catch(e => {
+    console.error(e)
+  })
+}
+
+const get2FASecret = (username, password) => {
+  return authenticateUser(username, password).then(user => {
+    if (!user) return null
+
+    const secret = otplib.authenticator.generateSecret()
+    const otpauth = otplib.authenticator.keyuri(username, 'Lamassu Industries', secret)
+    return { secret, otpauth }
+  })
+}
+
+const confirm2FA = (codeArg, context) => {
+  const code = codeArg
+  const requestingUser = context.req.session.user
+
+  if (!requestingUser) return false
+
+  return users.get2FASecret(requestingUser.id).then(user => {
+    const secret = user.twofa_code
+    const isCodeValid = otplib.authenticator.verify({ token: code, secret: secret })
+
+    if (!isCodeValid) return false
+    return true
+  })
+}
+
+const login = (username, password) => {
+  return authenticateUser(username, password).then(user => {
+    if (!user) return 'FAILED'
+    return users.get2FAMethod(user.id).then(user => {
+      return Promise.all([credentials.getHardwareCredentialsOfUser(user.id), user.twofa_code])
+    }).then(([devices, twoFASecret]) => {
+      if (!_.isEmpty(devices)) return 'FIDO2FA'
+      return twoFASecret ? 'INPUT2FA' : 'SETUP2FA'
+    })
+  })
+}
+
+const input2FA = (username, password, rememberMe, code, context) => {
+  return authenticateUser(username, password).then(user => {
+    if (!user) return false
+
+    return users.get2FASecret(user.id).then(user => {
+      const secret = user.twofa_code
+      const isCodeValid = otplib.authenticator.verify({ token: code, secret: secret })
+      if (!isCodeValid) return false
+
+      const finalUser = { id: user.id, username: user.username, role: user.role }
+      context.req.session.user = finalUser
+      if (rememberMe) context.req.session.cookie.maxAge = REMEMBER_ME_AGE
+
+      return true
+    })
+  })
+}
+
+const setup2FA = (username, password, secret, codeConfirmation) => {
+  return authenticateUser(username, password).then(user => {
+    if (!user || !secret) return false
+
+    const isCodeValid = otplib.authenticator.verify({ token: codeConfirmation, secret: secret })
+    if (!isCodeValid) return false
+
+    users.save2FASecret(user.id, secret)
+    return true
+  })
+}
+
+const getUserData = context => {
+  const lidCookie = context.req.cookies && context.req.cookies.lid
+  if (!lidCookie) return null
+
+  const user = context.req.session.user
+  return user
+}
+
+const validateRegisterLink = token => {
+  if (!token) return null
+  return users.validateUserRegistrationToken(token)
+    .then(r => {
+      if (!r.success) return null
+      return { username: r.username, role: r.role }
+    })
+    .catch(err => console.error(err))
+}
+
+const validateResetPasswordLink = token => {
+  if (!token) return null
+  return users.validatePasswordResetToken(token)
+    .then(r => {
+      if (!r.success) return null
+      return { id: r.userID }
+    })
+    .catch(err => console.error(err))
```

Comment:
same thing as above, .catch(console.error) also works
there are other places we can also change to this, if you decide to change